packages/codemods/src/codeMods/utilities/props.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  JsxAttribute,
+  JsxOpeningElement,
+  JsxSelfClosingElement,
+  SyntaxKind,
+  VariableDeclarationKind,
+  ts,
+  Node,
+} from 'ts-morph';
+import { EnumMap } from '../types';
+
+export function renameProp(
+  instances: (JsxOpeningElement | JsxSelfClosingElement)[],
+  toRename: string,
+  replacementName: string,
+  valueName: string | undefined,
+  changeValueMap: EnumMap | undefined,
+) {
+  instances.forEach(val => {
+    /* For each instance, first see if desired prop exists in the open. */
+
+    let foundProp = val.getAttribute(toRename);
+    if (foundProp) {
+      /* If found, do a simple name-replacementName. */
+      foundProp.set({ name: replacementName });
+      if (valueName) {
+        if (changeValueMap) {
+          /* If both are true, we have an enum and might have to change enum name AND value. */
+          let enumValue = (foundProp as JsxAttribute)
+            .getFirstChildByKind(SyntaxKind.JsxExpression)
+            .getFirstChildByKind(SyntaxKind.PropertyAccessExpression);
+          let firstEnumElem = enumValue.getFirstChildByKind(SyntaxKind.Identifier);
+          let secondEnumElem = enumValue.getLastChildByKind(SyntaxKind.Identifier);
+          firstEnumElem.rename(valueName);
+          secondEnumElem.rename(changeValueMap[secondEnumElem.getText()]);
+        } else {
+          /* If we don't have an enum, but we do have a value, only change the value of the prop to valueName. */
+          foundProp.set({ initializer: valueName });
```

Comment:
I'm not positive, but I suspect that this initializer might not quite work. At the very least, I think that the dev would need to know that they need to put in `FullEnumName.Value` otherwise it will resolve to `foo={<whatever was in valueName`

Reply:
You're right -- I think I wasn't explicit enough here. This isn't actually a case for enums, it's actually a case for full-on value change like a bool, where you need to flip the value!